src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -145,14 +150,27 @@ public void transform(StructuredRecord input, Emitter<KeyValue<StructuredRecord,
    */
   protected final void initOutput(BatchSinkContext context, BigQuery bigQuery, String outputName, String fqn,
                                   String tableName, @Nullable Schema tableSchema, String bucket,
-                                  FailureCollector collector, @Nullable String marker) throws IOException {
+                                  FailureCollector collector, @Nullable String marker,
+                                  Table table) throws IOException {
     LOG.debug("Init output for table '{}' with schema: {}", tableName, tableSchema);
 
     List<BigQueryTableFieldSchema> fields = BigQuerySinkUtils.getBigQueryTableFields(bigQuery, tableName, tableSchema,
       getConfig().isAllowSchemaRelaxation(), getConfig().getDatasetProject(),
       getConfig().getDataset(), getConfig().isTruncateTableSet(), collector);
 
     Configuration configuration = new Configuration(baseConfiguration);
+    if (table != null) {
+      com.google.cloud.bigquery.Schema bqSchema = table.getDefinition().getSchema();
+      if (bqSchema != null) {
+        configuration.set(BigQueryConstants.CONFIG_JSON_STRING_FIELDS,
+                BigQuerySinkUtils.getJsonStringFieldsFromBQSchema(bqSchema));
+        BigQuerySinkUtils.setJsonStringFields(fields, BigQuerySinkUtils.getJsonStringFieldsFromBQSchema(bqSchema));
```

Comment:
save the value from above call and pass here

Reply:
```java
if (bqSchema != null) {
  String jsonStringFields = BigQuerySinkUtils.getJsonStringFieldsFromBQSchema(bqSchema);
  configuration.set(BigQueryConstants.CONFIG_JSON_STRING_FIELDS, jsonStringFields);
  BigQuerySinkUtils.setJsonStringFields(fields, jsonStringFields);
}
```